handbook/engineering/distribution/gitlab_native_local.md

```diff
@@ -0,0 +1,59 @@
+# How to test the Gitlab native integration on a local dev machine
+
+This specializes and augments the official admin docs for the integration with gotchas when running on localhost.
+It assumes you want to develop/debug the Sourcegraph side of this relationship so therefore will run the `dev/start.sh`
+Sourcegraph and will run a Gitlab CE instance in a docker container.
+
+## Sourcegraph instance
+
+Launch a Sourcegraph instance as usual with `dev/start.sh` or `enterprise/dev/start.sh`.
+
+If you need a TLS endpoint for the Sourcegraph instance (for example to debug secure-only cookies) you can use [Caddy](https://caddyserver.com/),
+ [ngrok](https://dashboard.ngrok.com/get-started) or similar to create the TLS endpoint and reverse-proxy the Sourcegraph instance.
+ Please don't forget to update the external URL of the Sourcegraph instance in the site config to the appropriate TLS endpoint value.
+ 
+## Gitlab instance
+
+It is assumed here that you have a functioning Docker runtime on your local dev machine. We will run Gitlab like so:
+
+```shell script
+docker run --hostname localhost -p 4080:4080  \
+        --name gitlab \
+        --volume /Users/exampledev/sourcegraph-docker/gitlab-disk/config:/etc/gitlab \
+        --volume /Users/exampledev/sourcegraph-docker/gitlab-disk/logs:/var/log/gitlab \
+        --volume /Users/exampledev/sourcegraph-docker/gitlab-disk/data:/var/opt/gitlab \
+        gitlab/gitlab-ce:latest
+```
+
+> Important: Make sure if you run Gitlab on a port different from port 80 then the container port has to match the host port
+> so in this case 4080:4080.
+
+Once Gitlab is running it will set up configs in `/Users/exampledev/sourcegraph-docker/gitlab-disk/config`. We need to tell it
+its external URL, so go into that directory and open `gitlab.rb` in an editor. Uncomment and change the `external_url` value to `http://localhost:4080`.
+You have to restart Gitlab: `docker restart gitlab`.
+
+At this point you can add content to Gitlab to make it available for testing. Create a new project and clone the repo
```

Comment:
To make it easier to create new (and real) repositories (this is what I do for testing code intel on gitlab.k8ds.dev), I would suggest changing the steps to the below:

- Create the project in your GitLab instance, then copy the git `https` URL.
- Clone a repository from GitHub
- Change the `origin` remote URL of the cloned repository from GitHub to your local GItLab instance project using `git remote set-url origin <repo URL>`.
- Run `git push` to push the commits, tags, and branches to your GitLab project.

Reply:
@ryan-blunden That sounds great. Could you please make this change ? I haven't imported into gitlab before. Thank you :-)